File: solutions/elixir/darts/1/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    dist = :math.sqrt(x * x + y * y)

    cond do
      dist > 10 -> 0
      dist <= 1 -> 10
      dist <= 5 -> 5
      :default -> 1
    end
  end
end

File: solutions/haskell/darts/1/src/Darts.hs
module Darts (score) where

score :: Float -> Float -> Int
score x y
  | dist > 10 = 0
  | dist <= 1 = 10
  | dist <= 5 = 5
  | otherwise = 1
 where
  dist = sqrt $ (x * x) + (y * y)
